lib/matplotlib/collections.py

```diff
@@ -1462,7 +1462,7 @@ def __init__(self,
 
         .. plot:: gallery/lines_bars_and_markers/eventcollection_demo.py
         """
-
+        positions = np.array(positions, copy=True)
         segment = (lineoffset + linelength / 2.,
                    lineoffset - linelength / 2.)
         if positions is None or len(positions) == 0:
```

Comment:
At least this line suggests it was intended that `positions` could be None, but this PR changes the handling of None.

Reply:
Huh, you're right, even if there is a comment above (https://github.com/matplotlib/matplotlib/pull/14488/files#diff-506c6bd01694bddbd8999f2c6e920705R1408) that implies `positions` can't be `None`. I'll fix it to keep the current behaviour with None though (and add a smoke test for `positions=None`)

Reply:
I don't really know if it can be None or not, could probably use some cleanup :p

Reply:
Since passing `positions=None` would fail later on in some cases on `positions.sort()`, I've explicitly disallowed `None` input.

Reply:
It would previously not fail because the sort() is in another else clause; indeed `gca().add_collection(EventCollection(None))` works.
On the other hand there's basically not benefit to supporting None as input (in particular `eventplot(None)` already doesn't work) so I'm fine with dropping that, won't even insist on an api changes note... :)